problem/templates/problem/editTab/testcase.html

```diff
@@ -72,15 +56,75 @@
         </button>
         <h4 class='modal-title'>Edit Testcase</h4>
       </div>
-      <div class='modal-body'>
-        <p>Input</p>
-        <input type='file'><br>
-        <p>Output</p>
-        <input type='file'>
-      </div>
-      <div class='modal-footer'>
-        <button class='btn btn-info' data-dismiss='modal'>Save changes</button>
-      </div>
+      <form id="update_testcase" method="POST" 
+        enctype="multipart/form-data" action="{% url 'problem:problem.views.testcase' pid %}">
+        {% csrf_token %}
+        <div class='modal-body'>
+          <p>Input</p>
+          <input type='file' name='t_in' required><br>
+          <p>Output</p>
+          <input type='file' name='t_out' required>
+        </div>
+        <div class='modal-footer'>
+          <button class='btn btn-info'>Save changes</button>
+        </div>
+      </form>
     </div>
   </div>
 </div>
+
+<script>
+var update_tid = 0;
+$(document).ready(function() {
+    $("#add_testcase").submit(function (e) {
+        e.preventDefault();
+        add_new_testcase({{pid}}, new FormData(this));
+    });
+    $("#update_testcase").submit(function (e) {
+        e.preventDefault();
+        $.ajax({
+          type: "POST",
+          url: "/problem/{{pid}}/testcase/"+update_tid+"/",
+          data: new FormData(this),  
+          processData: false,
+          contentType: false,
+          success: function(data) {
+            alert("update success");
+          }
+        });
+    });
+    $(".reupload_btn").click(function (e) {
+        update_tid = $(this).parents("tr").attr('data-target');
+    });
+    $(".update_btn").click(function (e) {
+        var tid = $(this).parents("tr").attr('data-target');
+        var time = $("#"+tid+"_time").serialize();
+        var memory = $("#"+tid+"_memory").serialize();
+        var csrf = $("{% csrf_token %}").serialize();
+        $.ajax({
+            type: 'POST',
+            url: '/problem/'+{{pid}}+'/testcase/'+tid+'/',
+            data: time+"&"+memory+"&"+csrf,
+            success: function(data) {
+              alert('testcase updated')
+            }
+        });
+        return false;
+    });
+    $(".del_testcase_btn").click(function (e) {
+        var row = $(this).parents("tr");
+        var tid = $(this).parents("tr").attr('data-target');
+        $.ajax({
+            type: 'GET',
+            url: '/problem/'+{{pid}}+'/testcase/'+tid+'/delete/',
+            success: function(data) {
+              row.hide();
+            }
+        });
+        return false;
+    });
+});
+</script>
+{% else %}
+<strong>Save your problem before uploading testcases</strong>
+{% endif %}
```

Comment:
For overall of this file your ' and " usage may be same.
e.g. 
\<div class='panel panel-default' style='float:right;width:47%'>
and
\<div class="tab-pane" id="tag">
Their quotation usages are different.